Lib/test/test_httplib.py

```diff
@@ -2051,6 +2052,23 @@ def test_connect_with_tunnel(self):
         # This test should be removed when CONNECT gets the HTTP/1.1 blessing
         self.assertNotIn(b'Host: proxy.com', self.conn.sock.data)
 
+    def test_tunnel_connect_single_send_connection_setup(self):
+        """Regresstion test for https://bugs.python.org/issue43332."""
+        with mock.patch.object(self.conn, 'send') as mock_send:
+            self.conn.set_tunnel('destination.com')
+            self.conn.connect()
+            self.conn.request('GET', '/')
+        mock_send.assert_called()
+        # Likely 2, but this test only cares about the first.
+        self.assertGreater(
```

Comment:
Since we error message "unexpected number of send calls", the check for > 1 is a bit confusing to me. (This will fail with 0 or 1 call).   Checking for 2 calls directly and removing the comment "Likely 2" could be simpler. But I will leave it to you if have good reasons to think, this kind of check is better.

Reply:
I used >1 here as there should be 1 for the headers (the single send call we added) and at least one for the later actual http request itself.  i don't want to assert 2 because we aren't trying to test the behavior of the http traffic, just the initial proxy tunnel traffic.  (though it happens that the main http path was already optimized so it happens to be a single send of its own)